posthog/api/feature.py

```diff
@@ -94,8 +94,30 @@ def create(self, validated_data):
                 "payloads": {},
                 "multivariate": None,
             },
-        ).id
-        return super().create(validated_data)
+        )
+        validated_data["feature_flag_id"] = feature_flag.id
+
+        feature: Feature = super().create(validated_data)
+        feature_flag.filters = {
+            "groups": [
+                {
+                    "properties": [
+                        {
+                            "key": f"$feature_enrollment/{feature_flag_key}",
+                            "type": "person",
+                            "value": ["true"],
+                            "operator": "exact",
+                        }
+                    ],
+                    "rollout_percentage": 100,
+                    "feature_preview": str(feature.pk),
```

Comment:
suggestion: This makes things a bit brittle, since now there's a link from feature -> feature flag; and then a link back from feature_flag.filters -> feature.

I'd say generate this on the fly, or just do it on the frontend: We know the feature connected to the flag already (just like `experiment_set`) and can pass that to the frontend. Then, we search for properties with `$feature_enrollment` and make that the conditional check.

Reply:
Addressed!